openshift/console
# UI enhancement to support Quickstarts as CRs

# Issue
Addresses https://issues.redhat.com/browse/ODC-4751
and part of https://issues.redhat.com/browse/ODC-4752


**Solution**
Adds UI support for migration of QuickStarts to CRD


**How to Test the PR**
The QuickStart CRD and the CRs are placed in 
`frontend/packages/console-app/components/quickstarts/data/mock/yamls`

Step 1: ~~Apply `quick-start-crd.yaml` and wait for sometime for the server to update~~ (API now contains the CRD)
Step 2: Apply a few CRs present in individual YAML files in same directory.
Step 3: Check for the `QuickStartTile` in `+Add` page
Step 4: Check for `/quickstart` URL for the `QuickStart` catalog
Step 5: Click on a quickstart and check the `DrawerPanel`


# Screenshot
![qs-apply](https://user-images.githubusercontent.com/24852534/98534475-8f5db580-22aa-11eb-8b6c-03781f65eb6b.gif)
![qs-cata](https://user-images.githubusercontent.com/24852534/99506603-4f808780-29a8-11eb-83b8-7a4d3c21b40b.gif)


# Implementation
- [x] Converting QuickStarts to YAML
- [x] Script for installing QuickStart CRD and YAMLS for testing
- [x]  Moving existing QuickStart JSON to mocks
- [x]  Updating QuickStart types as per latest CRD structure
- [x] Fetching QuickStarts from API in QuickStart Tile
- [x] Fetching QuickStart from API in Drawer
- [x]  Updating components and utils as per latest type
- [x] Cleanup QuickStart utils which point to JSON ( currently unused )
- [x]  Serverless quickstart YAMLs  updated as it is currently being changed -- https://issues.redhat.com/browse/ODC-5059
- [x] Update exiting tests


# Tests 
Quick Start test suites updated

# Browser conformance
Chrome
